Zotero/Extensions/String+Extensions.swift

```diff
@@ -41,4 +41,9 @@ extension String {
         else { return nil }
         return ext.takeRetainedValue() as String
     }
+
+    var strippedHtmlTags: String {
+        guard !self.isEmpty else { return self }
+        return self.replacingOccurrences(of: "<[^>]*>", with: "", options: .regularExpression, range: nil)
```

Comment:
This is probably OK for notes, which are technically HTML and theoretically shouldn't have raw or unclosed angled brackets, but annotation comments aren't HTML — they just borrow HTML syntax for rich text. The first four HTML tags [here](https://www.zotero.org/support/kb/rich_text_bibliography) (with no attributes) are valid. `He had <10 dogs but definitely >2.` would be a valid comment, and you'd want to be able to find it with a search for `dogs`.

(The same applies to item fields, though in that case all of the tags on the linked page are supported.)

Reply:
> so for annotation comments I'll just remove i/b/sub/sup tags? 

Yes.

> Should I improve this for notes to only include tags we use too?

No, notes are actually HTML, so stripping all tags is appropriate.

> Do all item fields support those linked html tags?

Er, no, sorry, I guess just titles currently. It's possible HTML tags could be placed in Abstract by a translator, but they aren't technically HTML.

Reply:
@dstillman I don't support those tags in titles on iOS too yet. So do I just strip them or do I render them?

Reply:
If you have the ability to do rich text, it would be great to render those with the respective formatting, as we [just did for the web library](https://github.com/zotero/web-library/issues/515). See link there to the desktop code for this.

If you can't do rich text currently…I don't know. The desktop app always just showed the raw value before it started rendering the rich text.

For searching, strip.

Reply:
> Items list now renders those rich text tags

I'm not seeing anything that specifically handles the two supported `span` attributes. We shouldn't be rendering any other HTML.

> Do I maybe render rich text in item detail while not editing?

Yeah, you totally can. We just never got around to implementing that on desktop.

Reply:
@dstillman I used existing `HtmlAttributedStringConverter` (https://github.com/zotero/zotero-ios/blob/master/Zotero/Controllers/HtmlAttributedStringConverter.swift) which is used in annotation comments. 

But now I see it supports only `b`, `i`, `sup`, `sub`, so I'll add support for those 2 `span`s: https://github.com/zotero/zotero-ios/blob/f9bfbf88afe8b2630d135fbe04cabb1287543782/Zotero/Models/StringAttributes.swift#L22-L30